prow/cmd/hook/main.go

```diff
@@ -156,9 +154,9 @@ func main() {
 	}
 
 	var slackClient *slack.Client
-	if !o.dryRun && teamToken != "" {
+	if !o.dryRun && string(secretAgent.GetSecret(o.slackTokenFile)) != "" {
```

Comment:
```go
if !o.dryRun && o.slackTokenFile != "" {
```
is a bit cleaner

Reply:
Asking for a token that is not stored from the agent sounds like it should error out but as implemented this seems OK?